smt/managed/merklemanager.go

```diff
@@ -292,14 +300,33 @@ func (m *MerkleManager) Get(element int64) (Hash, error) {
 	return Hash(data).Copy(), nil
 }
 
-// GetNext
-// Get the next hash to be added to a state at this height
-func (m *MerkleManager) GetNext(element int64) (hash Hash) {
-	data, err := m.Manager.Get(m.key.Append("NextElement", element))
-	if err != nil {
-		return nil
+// GetIntermediate
+// Return the last two hashes that were combined to create the local
+// Merkle Root at the given index.  The element provided must be an
+// even power of two >= 2.
```

Comment:
> The element provided must be an even power of two >= 2.

I tried enforcing this, but adding any of the three if-statements below causes the tests to fail.

```go
if element < 4 {
	return nil, nil, fmt.Errorf("element is too small")
}
if bits.OnesCount64(uint64(element)) != 1 {
	return nil, nil, fmt.Errorf("element is not a power of 2")
}
if l := bits.Len64(uint64(element)) - 1; l%2 != 0 {
	return nil, nil, fmt.Errorf("element is not an even power of 2")
}
```

Reply:
The comment is wrong.  You can create a Merkle Root at any index.  I'll go fix it.